src/api/app/views/layouts/webui/_bottom_navigation.html.haml

```diff
@@ -7,7 +7,7 @@
           %span.d-block Watchlist
       %li.nav-item.border-start.border-gray-500
         = link_to(my_notifications_path, id: 'bottom-notifications-counter', class: 'nav-link text-light px-1 py-2', alt: 'Notifications') do
-          = render partial: 'layouts/webui/unread_notifications_counter'
+          = render partial: 'layouts/webui/unread_notifications_counter', locals: { unread_notifications: unread_notifications }
```

Comment:
`@unread_notifications`

Reply:
I'm also curious...

Reply:
Sorry I missed that we just pass this on. But then all of this makes little sense I fear.

`layouts/webui/webui.html.haml` get's a variable that contains a number from the controller to pass it on to  
`layouts/webui/top_navigation` and `layouts/webui/bottom_navigation` which both pass it on to `layouts/webui/unread_notifications_counter`.

That is a lot of code just to avoid to do `User.session ? User.session.unread_notifications : 0` in the `layouts/webui/unread_notifications_counter` partial. Which would equally only run the query only once as all of those happen in the same request cycle and the second render of `layouts/webui/unread_notifications_counter` would just hit the query cache...

Reply:
> Sorry I missed that we just pass this on. But then all of this makes little sense I fear.

Hm... :thinking:

> `layouts/webui/webui.html.haml` get's a variable that contains a number from the controller to pass it on to `layouts/webui/top_navigation` and `layouts/webui/bottom_navigation` which both pass it on to `layouts/webui/unread_notifications_counter`.

Yes, I agree, that's what we do with this pull request.

> That is a lot of code just to avoid to do `User.session ? User.session.unread_notifications : 0` in the `layouts/webui/unread_notifications_counter` partial. Which would equally only run the query only once as all of those happen in the same request cycle and the second render of `layouts/webui/unread_notifications_counter` would just hit the query cache...

We should rely on the query cache as less as possible, IMHO. Moreover, we could benefit from retrieving this value done by this SQL query on other pages, like the notifications page. So we don't need to rely on the query cache to retrieve this value, again.

Reply:
> We should rely on the query cache as less as possible, IMHO

Huh? Why?

Reply:
> > We should rely on the query cache as less as possible, IMHO
> 
> Huh? Why?

Performing repeated SQL queries which use the query cache is way more expensive than to pass one single value through our hierarchies of template partials.

The query cache is already there for cases like a user browsing a sequence of pages. We should not abuse using it.

By the way, I could add to this same pull request the code which will make use of the introduced `@unread_notifications` in the Notifications controller. Would that make it easier to see more advantages of the approach proposed?

Reply:
> Performing repeated SQL queries which use the query cache is way more expensive than to pass one single value through our hierarchies of template partials.

I doubt that as OBS developer time is probably the most expensive thing we deal with :)